source/server/broadcaster.cpp

```diff
@@ -36,50 +36,45 @@ void *StartBroadcasterThread(void *data) {
 
 Broadcaster::Broadcaster(Sequencer *sequencer) :
         m_sequencer(sequencer),
-        m_client_id(0),
-        m_socket(nullptr),
-        m_is_running(false),
+        m_client(nullptr),
         m_is_dropping_packets(false),
         m_packet_drop_counter(0),
         m_packet_good_counter(0) {
 }
 
-void Broadcaster::Start(int client_id, SWInetSocket *socket) {
-    m_client_id = client_id;
-    m_socket = socket;
-    m_is_running = true;
+void Broadcaster::Start(Client* client) {
+    m_client = client;
     m_is_dropping_packets = false;
     m_packet_drop_counter = 0;
     m_packet_good_counter = 0;
-
+    m_keep_running.store(true);
     m_msg_queue.clear();
 
     pthread_create(&m_thread, nullptr, StartBroadcasterThread, this);
 }
 
 void Broadcaster::Stop() {
-    m_queue_mutex.lock();
-    m_is_running = false;
-    m_queue_mutex.unlock();
-    m_queue_cond.signal();
-
-    pthread_join(m_thread, nullptr);
+    const bool was_running = m_keep_running.exchange(false);
+    if (was_running) {
+        m_queue_cond.signal();
+        pthread_join(m_thread, nullptr);
+    }
 }
 
 void Broadcaster::Thread() {
-    Logger::Log(LOG_DEBUG, "broadcaster m_thread %u owned by client_id %d", ThreadID::getID(), m_client_id);
-    while (m_is_running) {
+    Logger::Log(LOG_DEBUG, "Started broadcaster thread %u owned by client_id %d", ThreadID::getID(), m_client->GetUserId());
+    bool socket_error = false;
+    while (true) {

```

Comment:
```suggestion
    while (m_is_running) {
```

suggested by tritonas00 (https://github.com/RigsOfRods/ror-server/pull/90#issuecomment-498025809)

Reply:
`m_keep_running` not `m_is_running`